build-scripts/build_llvm.py

```diff
@@ -268,6 +283,10 @@ def main():
     )
     options = parser.parse_args()
 
+    # Ensure options.arch is always a list
+    if options.arch is None:
+        options.arch = []
+
```

Comment:
if required a list by default, it is better modify `parser.add_argument("--arch", nargs="+", ...`.  like `default=[]`.